ecs-agent/acs/session/task_manifest_responder.go

```diff
@@ -0,0 +1,177 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package session
+
+import (
+	"sync"
+
+	"github.com/aws/amazon-ecs-agent/ecs-agent/acs/model/ecsacs"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/metrics"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/wsclient"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type TaskManifestResponder interface {
+	New(taskComparer TaskComparer, messageIDAccessor ManifestMessageIDAccessor,
+		snAccessor SequenceNumberAccessor) wsclient.RequestResponder
+}
+
+type TaskComparer interface {
+	CompareRunningTasksOnInstanceWithManifest(*taskManifestResponder, *ecsacs.TaskManifestMessage) ([]*ecsacs.TaskIdentifier, error)
```

Comment:
Can we please add a description for this method for readers? Also this method takes a parameter of type `*taskManifestResponder` which is an unexported type. How would a struct implementing this interface outside this package get access to the unexported type?

Reply:
Synced offline, taskManifestResponder can be removed from this method's parameters as its only use was logging. `ecs-agent/logging` can be used to that end.